packager/file/CMakeLists.txt

```diff
@@ -29,6 +29,10 @@ target_link_libraries(file
     status
     version)
 
+if(LIBPACKAGER_SHARED)
+    target_compile_definitions(file PUBLIC SHAKA_IMPLEMENTATION)
```

Comment:
wince we have a SHAKA_EXTERNAL definition in file I think this is needed

Reply:
moved it to macros.h, trying to include packager.h would be more complicated due to everything else it defines